### ISSUE TITLE
Undefine symbol 

### ISSUE THREAD

**Issue by bigbrett** [ThrowTheSwitch/Ceedling#552]
# TL;DR:
**Is there any good way to UNDEFINE system macros globally for a ceedling project? Or maybe just on a test-by-test basis? 

Bonus question: Has anyone been able to successfully leverage ceedling/CMock/Unity for testing a project using the Nordic NRF5 SDK? What wizardry did you use? All these complicated macro shim layers and redirections  in the SDK are causing ceedling to have a stroke and I still can't figure it out

# Background

Nordic Semiconductor has peppered their SDK with #ifdef guards against the target platform for some god forsaken reason, as shown below

```
#ifndef __unix__
#include "IMPT_CHIP_DEFINITIONS.h"
#endif 
```
Unfortunately, this file is included by almost everything in their driver layer, which makes integration with ceedling a nightmare (among other reasons). Small contrived example: an app module that does stuff with the on-chip RTC would have an include chain something like:

```                                                                                           
[my_app_module.c] --> [nrf_rtc_driver] --> [nrf_driver_backend] --> [nrf.h] ------(#ifndef __unix__)------> all the impt chip definitions (incl CMSIS header w/ register defs, peripherals, IRQ numbers, etc.....)
```

I really just need to mock `nrf_rtc_driver`, but since it uses a bunch of crap behind the stupid ifndef guards, I can't get the basic macros that need to be defined. To redefine them all manually in the `project.yml` file would be thousands and thousands of lines. 

**Is there any good way to UNDEFINE system macros globally for a ceedling project? Or maybe just on a test-by-test basis?**

**Bonus question: Has anyone been able to successfully leverage ceedling/CMock/Unity for testing a project using the Nordic NRF5 SDK? What wizardry did you use? All these complicated macro shim layers and redirections  in the SDK are causing ceedling to have a stroke and I still can't figure it out**
